notebooks/304-pytorch-post-training-quantization/304-pytorch-post-training-quantization.ipynb

```diff
@@ -0,0 +1,714 @@
+    "            acc1, acc5 = accuracy(output, target, topk=(1, 5))\n",
+    "            top1.update(acc1[0], images.size(0))\n",
+    "            top5.update(acc5[0], images.size(0))\n",
+    "\n",
+    "            # measure elapsed time\n",
+    "            batch_time.update(time.time() - end)\n",
+    "            end = time.time()\n",
+    "\n",
+    "            print_frequency = 10\n",
+    "            if i % print_frequency == 0:\n",
+    "                progress.display(i)\n",
+    "\n",
+    "        print(' * Acc@1 {top1.avg:.3f} Acc@5 {top5.avg:.3f}'\n",
+    "              .format(top1=top1, top5=top5))\n",
+    "    return top1.avg"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1912cb42",
+   "metadata": {},
+   "source": [
+    "#### Create and load original uncompressed model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3a92b300",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def create_model(model_path):\n",
+    "    model = models.resnet50(pretrained=True)\n",
+    "    # update the last FC layer for Tiny ImageNet number of classes\n",
+    "    NUM_CLASSES = 200\n",
+    "    model.fc = nn.Linear(in_features=2048, out_features=NUM_CLASSES, bias=True)\n",
+    "    model.to(device)\n",
+    "    if model_path.exists():\n",
+    "        checkpoint = torch.load(str(model_path), map_location='cpu')\n",
+    "        model.load_state_dict(checkpoint['state_dict'], strict=True)\n",
+    "    else:\n",
+    "        raise RuntimeEror('There is no checkpoint to load')\n",
+    "    return model\n",
+    "\n",
+    "\n",
+    "model = create_model(MODEL_DIR / fp32_checkpoint_filename)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4291ed2f",
+   "metadata": {},
+   "source": [
+    "#### Create train and validation dataloaders "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "93213778",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def create_dataloaders(batch_size=128):\n",
+    "    train_dir = DATASET_DIR / 'train'\n",
+    "    val_dir = DATASET_DIR / 'val' / 'images'\n",
+    "    normalize = transforms.Normalize(mean=[0.485, 0.456, 0.406],\n",
+    "                                     std=[0.229, 0.224, 0.225])\n",
+    "    train_dataset = datasets.ImageFolder(\n",
+    "        train_dir,\n",
+    "        transforms.Compose([\n",
+    "            transforms.Resize(IMAGE_SIZE),\n",
+    "            transforms.ToTensor(),\n",
+    "            normalize,\n",
+    "        ]))\n",
+    "    val_dataset = datasets.ImageFolder(\n",
+    "        val_dir,\n",
+    "        transforms.Compose([\n",
+    "            transforms.Resize(IMAGE_SIZE),\n",
+    "            transforms.ToTensor(),\n",
+    "            normalize\n",
+    "        ]))\n",
+    "\n",
+    "    train_loader = torch.utils.data.DataLoader(train_dataset, batch_size=batch_size, shuffle=True,\n",
```

Comment:
the main point of this notebook that it doesn't use training pipeline. is it possible to use only validation set to get a good initialization. AFAIKK, POT uses calibration dataset for initialization and works fine.